services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__patch.py

```diff
@@ -0,0 +1,226 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=too-many-arguments
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-statements
+
+
+import json
+from http import HTTPStatus
+
+import pytest
+from aiohttp.test_utils import TestClient
+from pytest_mock.plugin import MockerFixture
+from pytest_simcore.helpers.utils_assert import assert_status
+from pytest_simcore.helpers.utils_login import UserInfoDict
+from servicelib.aiohttp import status
+from simcore_service_webserver._meta import api_version_prefix
+from simcore_service_webserver.db.models import UserRole
+from simcore_service_webserver.projects.models import ProjectDict
+
+API_PREFIX = "/" + api_version_prefix
+
+
+@pytest.fixture
+def mock_catalog_api_get_services_for_user_in_product(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.get_services_for_user_in_product",
+        spec=True,
+        return_value=[],
+    )
+
+
+@pytest.fixture
+def mock_project_uses_available_services(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.project_uses_available_services",
+        spec=True,
+        return_value=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "user_role,expected",
+    [
+        (UserRole.ANONYMOUS, status.HTTP_401_UNAUTHORIZED),
+        (UserRole.GUEST, status.HTTP_204_NO_CONTENT),
+        (UserRole.USER, status.HTTP_204_NO_CONTENT),
+        (UserRole.TESTER, status.HTTP_204_NO_CONTENT),
+        (UserRole.ADMIN, status.HTTP_204_NO_CONTENT),
+        (UserRole.PRODUCT_OWNER, status.HTTP_204_NO_CONTENT),
+    ],
+)
+async def test_patch_project_access_rights(
```

Comment:
in order to not confuse I would rename that test.
projects access rights is something we have in the backend. what you are testing here is accessibility of the entrypoints based on user role and which is static. maybe rename to 
```suggestion
async def test_patch_project_entrypoint_access(
```